examples/pr_notifier/notifier.py

```diff
@@ -54,28 +57,50 @@ class Developer(JSONWizard):
     """Generic representation of a user from GitHub/GitLab."""
 
     name: str
+    email: str | None
     avatar_url: str | None
 
     @staticmethod
     def from_github(user: dict[str, Any]):
         """Creates from a GitHub user"""
-        return Developer(name=user["login"], avatar_url=user["avatar_url"])
+        return Developer(name=user["login"], email=None, avatar_url=user["avatar_url"])
 
     @staticmethod
     def from_gitlab(user: dict[str, Any]):
         """Creates from a GitLab user"""
-        return Developer(name=user["username"], avatar_url=user["avatar_url"])
+        email = user["username"] + GITLAB_EMAIL_DOMAIN if GITLAB_EMAIL_DOMAIN else None
+        return Developer(
+            name=user["username"], email=email, avatar_url=user["avatar_url"]
+        )
+
+    @functools.lru_cache(maxsize=10)
+    def get_slack_username(self) -> str:
+        """Gets the name that should be used on Slack."""
+        if not self.email:
+            return self.name
```

Comment:
This is the same point about newlines that I mentioned before but didn't find a good linting rule for. As a rule, I would put a newline after the end of each indentation, so in this case:

```
@functools.lru_cache(maxsize=10)
    def get_slack_username(self) -> str:
        """Gets the name that should be used on Slack."""
        if not self.email:
            return self.name
        
        response = client.users_lookupByEmail(email=self.email)
        if not response["ok"]:
            logging.error("Getting slack id for %s: %s", self.name, response["error"])
            return self.name
            
        return f'<@{response["user"]["id"]}>'  # type: ignore
```

Reply:
Sorry about that, I will try and pay more attention to this. It's much cleaner with some breathing room.